Mathlib/Geometry/Manifold/MFDeriv/Basic.lean

```diff
@@ -1038,31 +1038,44 @@ theorem MDifferentiableWithinAt.mfderivWithin_mono_of_mem_nhdsWithin
     mfderivWithin I I' f t x = mfderivWithin I I' f s x :=
   (HasMFDerivWithinAt.mono_of_mem_nhdsWithin h.hasMFDerivWithinAt h₁).mfderivWithin hxt
 
-theorem Filter.EventuallyEq.mfderivWithin_eq (hs : UniqueMDiffWithinAt I s x) (hL : f₁ =ᶠ[𝓝[s] x] f)
-    (hx : f₁ x = f x) : mfderivWithin I I' f₁ s x = mfderivWithin I I' f s x := by
+theorem Filter.EventuallyEq.mfderivWithin_eq (hL : f₁ =ᶠ[𝓝[s] x] f) (hx : f₁ x = f x) :
+    mfderivWithin I I' f₁ s x = mfderivWithin I I' f s x := by
   by_cases h : MDifferentiableWithinAt I I' f s x
-  · exact (h.hasMFDerivWithinAt.congr_of_eventuallyEq hL hx).mfderivWithin hs
+  · unfold mfderivWithin
+    simp only [h, (hL.mdifferentiableWithinAt_iff hx).1 h, ↓reduceIte, writtenInExtChartAt]
+    apply Filter.EventuallyEq.fderivWithin_eq; swap
+    · simp [hx]
+    filter_upwards [extChartAt_preimage_mem_nhdsWithin (I := I) hL] with y hy
+    simp only [preimage_setOf_eq, mem_setOf_eq] at hy
+    simp [-extChartAt, hy, hx]
```

Comment:
The `-extChartAt` is a small code smell --- but I see why `extChartAt` is marked simp; perhaps this is the locally optimal proof.